pipelines/tekton/aiedge-e2e/tasks/retrieve-image-digest.task.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: retrieve-image-digest
+spec:
+  description: Record location and SHA of the built image
+  params:
+  - name: namespace
+    type: string
+  - name: model-name
+    type: string
+  - name: model-version
+    type: string
+  - name: target-registry-url
+    type: string
+  steps:
+  - name: get-image-sha
+    image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+    script: |
+      echo -n "$(params.target-registry-url)" | tee $(results.target-registry-url.path) ;
+      echo ;
+      oc get -n $(params.namespace) -o jsonpath='{.status.tags[?(@.tag == "$(params.model-version)")].items[0].image}' imagestream/$(params.model-name) | tee $(results.image-sha.path)
+  results:
+  - name: target-registry-url
+    description: The target-registry-url where the image will be stored
+  - name: image-sha
+    description: The image checksum
```

Comment:
Since this PR merges the build and test pipeline, you can probably delete this task and just grab the url and sha from the `build-container` step since it uses `buildah` [which emits those as results already](https://github.com/tektoncd/catalog/blob/d0f77dabc071508255bd09fbc56c128efe99b226/task/buildah/0.6/buildah.yaml#L66-L70).

Reply:
I just tested it and the SHA is the same, but the return registry url is the internal one (which is expected). 
```
pipelineResults:
  - name: target-registry-url
    value: quay.io/pplaczek/tensorflow-housing
  - name: image-sha
    value: 'sha256:108ae6c1b496942f13f2da9ead5b40611f097ca5e9d6b3eec9a293b54c60f344'
  - name: buildah-registry-url
    value: >
      image-registry.openshift-image-registry.svc:5000/pplaczek-pipeline-dev/tensorflow-housing:1
  - name: buildah-sha
    value: 'sha256:108ae6c1b496942f13f2da9ead5b40611f097ca5e9d6b3eec9a293b54c60f344'
```
Still, this task pretty much just copies a parameter into a result, so that could be done in the pipeline itself.

Reply:
Nevermind, looks like the task _is_ necessary 😅. I thought you could at least do basic param substitution but apparently not.
```
#  - name: target-registry-url
#   value: quay.io/$(params.target-imagerepo)/$(params.model-name)
    
admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: expected pipeline results to be task result expressions but an invalid expressions was found: spec.results[7].value
invalid value: referencing a nonexistent task: spec.results[7].value
```

Reply:
Ok. After this PR, we can follow-up with some refactors once we have a basic test in place to verify the functionality of this pipeline